internal/connectors/plugins/public/column/utils.go

```diff
@@ -0,0 +1,239 @@
+		return fmt.Errorf("required field amount must be provided")
+	}
+
+	if pi.SourceAccount == nil {
+		return fmt.Errorf("required field sourceAccount must be provided")
+	}
+
+	if pi.SourceAccount.Reference == "" {
+		return fmt.Errorf("required field sourceAccount.reference must be provided")
+	}
+
+	if pi.DestinationAccount == nil {
+		return fmt.Errorf("required field destinationAccount must be provided")
+	}
+
+	if pi.DestinationAccount.Reference == "" {
+		return fmt.Errorf("required field destinationAccount.reference must be provided")
+	}
+
+	if pi.Metadata == nil {
+		return fmt.Errorf("required field metadata must be provided")
+	}
+
+	payoutType := models.ExtractNamespacedMetadata(pi.Metadata, client.ColumnPayoutTypeMetadataKey)
+
+	if payoutType == "" {
+		return fmt.Errorf("required field metadata field %s must be provided", client.ColumnPayoutTypeMetadataKey)
+	}
+
+	if payoutType != "ach" && payoutType != "wire" && payoutType != "realtime" && payoutType != "international-wire" {
+		return fmt.Errorf("required field metadata field %s must be one of: ach, wire, realtime, international-wire", client.ColumnPayoutTypeMetadataKey)
+	}
+
+	if pi.Asset == "" {
+		return fmt.Errorf("required field asset must be provided")
+	}
+
+	return nil
+}
+
+func (p *Plugin) validateReversePayout(pr models.PSPPaymentInitiationReversal) error {
+	if pr.Metadata == nil {
+		return fmt.Errorf("required field metadata must be provided")
+	}
+
+	reason := models.ExtractNamespacedMetadata(pr.Metadata, client.ColumnReasonMetadataKey)
+	if reason == "" {
+		return fmt.Errorf("required field metadata field %s must be provided", client.ColumnReasonMetadataKey)
+	}
+
+	if !IsValidReversePayoutReason(reason) {
+		return fmt.Errorf("required field metadata field %s must be a valid reason", client.ColumnReasonMetadataKey)
+	}
+
+	if pr.RelatedPaymentInitiation.Reference == "" {
+		return fmt.Errorf("required field relatedPaymentInitiation.reference must be provided")
+	}
+
+	return nil
+}
+
+func validateExternalBankAccount(newExternalBankAccount models.BankAccount) error {
+	if newExternalBankAccount.AccountNumber == nil {
+		return fmt.Errorf("account number is required")
+	}
+
+	routingNumber := models.ExtractNamespacedMetadata(newExternalBankAccount.Metadata, client.ColumnRoutingNumberMetadataKey)
+
+	if routingNumber == "" {
+		return fmt.Errorf("required metadata field %s is missing", client.ColumnRoutingNumberMetadataKey)
+	}
+
+	err := validateAddress(newExternalBankAccount.Metadata)
+
+	if err != nil {
+		return err
+	}
+
+	addressLine1 := models.ExtractNamespacedMetadata(newExternalBankAccount.Metadata, client.ColumnAddressLine1MetadataKey)
+
+	if addressLine1 != "" {
+
+		if newExternalBankAccount.Country == nil {
+			return fmt.Errorf("country is required")
+		}
+
+	}
+
+	if addressLine1 == "" && newExternalBankAccount.Country != nil {
+		return fmt.Errorf("metadata field country is not required when addressLine1 is not provided")
+	}
```

Comment:
As far as I can tell from the documentation the following fields should always be present in the Bank Account creation request:
* address.line_1
* address.city
* address.country_code

So I think we can simply check if all these fields are present, and if not return a validation error. The split of validation logic here + the validation logic in `validateAddress` is not optimal in my opinion as it creates a double validation and makes the conditions in the code harder to understand and therefore easier to create bugs.

Can we simplify it so that the validation function is the one that returns the errors? Perhaps it could also extract address data into an intermediate `Address` struct that can be be used in the requests to both the BankAccount creation and Payout initiation requests.